### ISSUE TITLE
Complete codegen and add tests

### ISSUE THREAD

**Issue by grjte** [0xPolygonMiden/air-dsl#9]
The initial codegen may be missing some things in order to produce valid code, since it was primarily targeting generation of constraints.

The following should be added:
- any missing required codegen in order for output files to be valid
- Unit tests for each of the generated methods / structs
- tests for generation of entire files

We may want to consider something like [expect_test](https://docs.rs/expect-test/latest/expect_test/) for this (maybe for IR and parsing as well)

**Comment by grjte** [comment]
This can be started once #8 is merged

**Comment by grjte** [comment]
@tohrnii since there are now multiple other issues where the remaining tasks are for codegen (#10, #11, #14, #15 ), I'm renaming this one to be specifically for unit testing, and I'm going to unassign it from you for now, since it's lower priority than the other items we need to finish.
